File: Hilos/hilos.go
package main

import (
	"fmt"
	"time"
	"sync"
)

func es_primo(num int) bool {
	for i := 2; i < (num/2)+1; i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

func numeros_primos(inicio,fin int, wg *sync.WaitGroup) {
	defer wg.Done()
	for i := inicio; i < fin; i++ {
		if es_primo(i){
			fmt.Println(" El numero", i, "es primo")
		}
	}
}

func main() {
    var wg sync.WaitGroup
    inicio := time.Now()

    wg.Add(1)
    go numeros_primos(1, 100000, &wg)

    wg.Wait()
    fin := time.Now()
    fmt.Printf("Tiempo: %v\n", fin.Sub(inicio))
}

File: Hilos/hilos.py
import threading
import time

def es_primo(num):
    for i in range(2, (num // 2) + 1):
        if num % i == 0:
            return False
    return True

def numeros_primos(inicio, fin):
    for i in range(inicio, fin):
        if es_primo(i):
            print("El numero", i, "es primo")

def main():
    inicio_tiempo = time.time()

    hilo1 = threading.Thread(target=numeros_primos, args=(1, 100000))

    hilo1.start()

    hilo1.join()

    fin_tiempo = time.time()
    print("Tiempo:", fin_tiempo - inicio_tiempo)

if __name__ == "__main__":
    main()